### PR TITLE
[RDY] Window scaling for high dpi support

### PR DESCRIPTION
**Pull request by TheCycoONE**
New scaling option in the config allows the window size to be a multiple of the logical resolution.
This addressed #1210

### CODE REVIEW

**Comment by Alberth289346** [comment]
While testing, it appears that in the settings.
- A "scale" parameter was added, and
- Size of the screen is interpreted as unscaled size.

Please extend the textual description how to interpret the scale and the screen size here.

**Comment by TheCycoONE** [comment]
I've updated the description of the various window size / resolution settings. I want to properly expose scale in the settings page as well before this is merged.

File: CorsixTH/Src/th_gfx_sdl.cpp [hunk 1]
```diff
@@ -232,7 +232,7 @@ bool THRenderTarget::create(const THRenderTargetCreationParams* pParams)
     m_pFormat = SDL_AllocFormat(SDL_PIXELFORMAT_ABGR8888);
     m_pWindow = SDL_CreateWindow("CorsixTH",
                                  SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED,
-                                 pParams->iWidth, pParams->iHeight,
+                                 pParams->iWidth * pParams ->fWindowScale, pParams->iHeight * pParams->fWindowScale,
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx_sdl.cpp]: Spurious space:
```` c
pParams ->fWindowScale
       ^
````

File: CorsixTH/Lua/dialogs/resizables/options.lua [hunk 2]
```diff
@@ -149,6 +164,8 @@ local --[[persistable:options_window_language_button]] function language_button(
 local --[[persistable:options_width_textbox_reset]] function width_textbox_reset() end
 local --[[persistable:options_height_textbox_reset]] function height_textbox_reset() end
 
+
+
```
Comment (Alberth289346) [inline CorsixTH/Lua/dialogs/resizables/options.lua]: Any particular reason for adding empty lines?
While at it, just write a line documentation perhaps?

Comment (TheCycoONE) [inline CorsixTH/Lua/dialogs/resizables/options.lua]: No, an oversight. I was going to write the callback function here, but I needed the closure.

**Comment by TheCycoONE** [comment]
I have doubts about making committing these changes without:
a) An auto mode that is default
b) and HighDPI awareness in Windows

As is I don't think this feature brings any improvement to the situation on Windows and if we release with a scale of 1 and add auto later a lot of people will end up with tiny windows and have to edit their config

**Comment by Alberth289346** [comment]
good point

**Comment by TheCycoONE** [comment]
SDLs scaling is not ideal for our use.

**Comment by Alberth289346** [comment]
In OpenGL you can configure interpolation with GL_LINEAR, see "Texture Filtering" at https://learnopengl.com/#!Getting-started/Textures . If SDL uses opengl for this, it may be configurable too

**Comment by TheCycoONE** [comment]
It's slightly configurable: https://wiki.libsdl.org/SDL_HINT_RENDER_SCALE_QUALITY?highlight=(\bCategoryDefine\b)|(CategoryHints)

**Comment by Alberth289346** [comment]
0 or "nearest" looks like very much "just take the pixel". OpenGL also names it "nearest". Does that do anything useful? (In the sense that the white of the letters should stay white, and not get interpolated or anti-aliased with background (which the OP sees as blurring, I think).)

**Comment by TheCycoONE** [comment]
@reubot are you able to test / take over my PR then?

**Pull request closed without merging**